tgui/packages/tgui/interfaces/Phone.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file
+ * @copyright 2022
+ * @author DisturbHerb (https://github.com/disturbherb)
+ * @license MIT
+ */
+
+import { useBackend } from '../backend';
+import { Button, Collapsible, Dimmer, Section } from '../components';
+import { Window } from '../layouts';
+
+export const Phone = (props, context) => {
+  const { data } = useBackend(context);
+  const { inCall, dialing, name } = data;
+  return (
+    <Window title={name} width={250} height={350}>
+      <Window.Content>
+        {(dialing || inCall) && (
+          <Dimmer>
+            <h1>LINE BUSY</h1>
+          </Dimmer>
+        )}
+        <Section title="Phonebook" fill scrollable>
+          <AddressGroup category="bridge" name="Bridge" depColour="green" />
+          <AddressGroup category="engineering" name="Engineering" depColour="yellow" />
+          <AddressGroup category="medical" name="Medical" depColour="blue" />
+          <AddressGroup category="research" name="Research" depColour="purple" />
+          <AddressGroup category="security" name="Security" depColour="red" />
+          <AddressGroup category="uncategorized" name="Uncategorized" depColour="brown" />
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const AddressGroup = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { category, name, depColour } = props;
+  const { phonebook } = data;
```

Comment:
Possibly better to have this component receive a slice of the phonebook, rather than use it from the backend directly? No big deal, but if you did e.g. a single `data.reduce` over in the `Phone` component to create a map of arrays to pull from it'd likely read a little neater.

Reply:
I can't mark an arbitrarily large amount of code with a suggestion, so broadly including thoughts below here:

For `Phone` component, include `phonebook` in `data` destructuring, then:
```
const addressGroups = phonebook.reduce((acc, curr) => {
  const { category } = curr;
  if (acc[category] === undefined) {
    acc[category] = [];
  }
  acc[category].push(curr);
  return acc;
}, {});
```
This builds up a dictionary that is something like:
```
{
  bridge: [{/* some bridge phone */}, {/* some other bridge phone */},
  engineering: [{/* some engineering phone */}],
  // other categories
}
```
In the rendering part, you then do:
```
<Section title="Phonebook" fill scrollable>
  <AddressGroup name="Bridge" depColour="green" phones={addressGroups['bridge']} />
  <AddressGroup name="Engineering" depColour="yellow" phones={addressGroups['engineering']} />
  // other groups
</Section>
```
And then in the `AddressGroup` rather than pulling from `data` you just use the `phones` prop you're given.